src/main/java/uk/gov/pay/ledger/event/resource/EventResource.java

```diff
@@ -58,6 +66,22 @@ public Event getEvent(@PathParam("eventId") Long eventId) {
                 .orElseThrow(() -> new WebApplicationException(Response.Status.NOT_FOUND));
     }
 
+    @POST
+    @Timed
+    public Response writeEvent(List<EventMessageDto> events) throws QueueException {
+        try {
+            eventMessageHandler.processEventBatch(events
+                    .stream()
+                    .map(eventMessageDto -> EventMessage.of(eventMessageDto, null))
+                    .collect(Collectors.toList())
```

Comment:
Please use the (annoyingly verbose) `Collectors.toUnmodifiableList()` if it’s not going to be modified.